client/client.go

```diff
@@ -478,14 +478,20 @@ func (r *NotaryRepository) GetTargetByName(name string, roles ...string) (*Targe
 
 }
 
+// TargetSignedStruct is a struct that contains information about the Target and the list of signatures
+type TargetSignedStruct struct {
```

Comment:
Non-blocking - this is just my misunderstanding of what we wanted:  did we want to also include the role in here so that we can look up any x509 certificates too?